--- a/definitions.unpick
+++ b/definitions.unpick
@@ -13,174 +_,14 @@
 
 target_method net.minecraft.server.level.TicketType register (Ljava/lang/String;JI)Lnet/minecraft/server/level/TicketType;
 	param 1 chunk_ticket_type_expiration
-	param 2 chunk_ticket_type_flags
 
 target_method net.minecraft.server.level.TicketType <init> (JI)V
 	param 0 chunk_ticket_type_expiration
-	param 1 chunk_ticket_type_flags
 
 target_method net.minecraft.server.level.TicketType timeout ()J
 	return chunk_ticket_type_expiration
 
-target_method net.minecraft.server.level.TicketType flags ()I
-	return chunk_ticket_type_flags
-
-#: Integers in ARGB color format.
-group int argb_color
-	@format hex
-	net.minecraft.util.CommonColors.*
-	net.minecraft.realms.RealmsScreen.COLOR_DARK_GRAY
-	net.minecraft.realms.RealmsScreen.COLOR_MEDIUM_GRAY
-	net.minecraft.realms.RealmsScreen.COLOR_GREEN
-	net.minecraft.realms.RealmsScreen.COLOR_LINK
-	net.minecraft.realms.RealmsScreen.COLOR_LINK_HOVER
-
-target_method net.minecraft.util.ARGB alpha (I)I
-	param 0 argb_color
-
-target_method net.minecraft.util.ARGB red (I)I
-	param 0 argb_color
-
-target_method net.minecraft.util.ARGB green (I)I
-	param 0 argb_color
-
-target_method net.minecraft.util.ARGB blue (I)I
-	param 0 argb_color
-
-target_method net.minecraft.util.ARGB color (IIII)I
-	return argb_color
-
-target_method net.minecraft.util.ARGB color (III)I
-	return argb_color
-
-target_method net.minecraft.util.ARGB color (Lnet/minecraft/world/phys/Vec3;)I
-	return argb_color
-
-target_method net.minecraft.util.ARGB multiply (II)I
-	param 0 argb_color
-	param 1 argb_color
-	return argb_color
-
-target_method net.minecraft.util.ARGB scaleRGB (IF)I
-	param 0 argb_color
-	return argb_color
-
-target_method net.minecraft.util.ARGB scaleRGB (II)I
-	param 0 argb_color
-	return argb_color
-
-target_method net.minecraft.util.ARGB greyscale (I)I
-	param 0 argb_color
-	return argb_color
-
-target_method net.minecraft.util.ARGB srgbLerp (FII)I
-	param 1 argb_color
-	param 2 argb_color
-	return argb_color
-
-target_method net.minecraft.util.ARGB opaque (I)I
-	param 0 argb_color
-	return argb_color
-
-target_method net.minecraft.util.ARGB transparent (I)I
-	param 0 argb_color
-	return argb_color
-
-target_method net.minecraft.util.ARGB color (II)I
-	param 1 argb_color
-	return argb_color
-
-target_method net.minecraft.util.ARGB color (FI)I
-	param 1 argb_color
-	return argb_color
-
-target_method net.minecraft.util.ARGB white (F)I
-	return argb_color
-
-target_method net.minecraft.util.ARGB colorFromFloat (FFFF)I
-	return argb_color
-
-target_method net.minecraft.util.ARGB average (II)I
-	param 0 argb_color
-	param 1 argb_color
-	return argb_color
-
-target_method net.minecraft.util.ARGB toABGR (I)I
-	param 0 argb_color
-
-target_method net.minecraft.util.ARGB fromABGR (I)I
-	return argb_color
-
-target_method net.minecraft.client.gui.Font drawInBatch (Ljava/lang/String;FFIZLorg/joml/Matrix4f;Lnet/minecraft/client/renderer/MultiBufferSource;Lnet/minecraft/client/gui/Font$DisplayMode;II)V
-	param 3 argb_color
-
-target_method net.minecraft.client.gui.Font drawInBatch (Lnet/minecraft/network/chat/Component;FFIZLorg/joml/Matrix4f;Lnet/minecraft/client/renderer/MultiBufferSource;Lnet/minecraft/client/gui/Font$DisplayMode;II)V
-	param 3 argb_color
-
-target_method net.minecraft.client.gui.Font drawInBatch (Lnet/minecraft/util/FormattedCharSequence;FFIZLorg/joml/Matrix4f;Lnet/minecraft/client/renderer/MultiBufferSource;Lnet/minecraft/client/gui/Font$DisplayMode;II)V
-	param 3 argb_color
-
-target_method net.minecraft.client.gui.Font drawInBatch8xOutline (Lnet/minecraft/util/FormattedCharSequence;FFIILorg/joml/Matrix4f;Lnet/minecraft/client/renderer/MultiBufferSource;I)V
-	param 3 argb_color
-	param 4 argb_color
-
-target_method net.minecraft.client.gui.GuiGraphics drawCenteredString (Lnet/minecraft/client/gui/Font;Ljava/lang/String;III)V
-	param 4 argb_color
-
-target_method net.minecraft.client.gui.GuiGraphics drawCenteredString (Lnet/minecraft/client/gui/Font;Lnet/minecraft/network/chat/Component;III)V
-	param 4 argb_color
-
-target_method net.minecraft.client.gui.GuiGraphics drawCenteredString (Lnet/minecraft/client/gui/Font;Lnet/minecraft/util/FormattedCharSequence;III)V
-	param 4 argb_color
-
-target_method net.minecraft.client.gui.GuiGraphics drawString (Lnet/minecraft/client/gui/Font;Ljava/lang/String;III)V
-	param 4 argb_color
-
-target_method net.minecraft.client.gui.GuiGraphics drawString (Lnet/minecraft/client/gui/Font;Ljava/lang/String;IIIZ)V
-	param 4 argb_color
-
-target_method net.minecraft.client.gui.GuiGraphics drawString (Lnet/minecraft/client/gui/Font;Lnet/minecraft/network/chat/Component;III)V
-	param 4 argb_color
-
-target_method net.minecraft.client.gui.GuiGraphics drawString (Lnet/minecraft/client/gui/Font;Lnet/minecraft/network/chat/Component;IIIZ)V
-	param 4 argb_color
-
-target_method net.minecraft.client.gui.GuiGraphics drawWordWrap (Lnet/minecraft/client/gui/Font;Lnet/minecraft/network/chat/FormattedText;IIIIZ)V
-	param 5 argb_color
-
-target_method net.minecraft.client.gui.GuiGraphics drawWordWrap (Lnet/minecraft/client/gui/Font;Lnet/minecraft/network/chat/FormattedText;IIII)V
-	param 5 argb_color
-
-target_method net.minecraft.client.gui.GuiGraphics hLine (IIII)V
-	param 3 argb_color
-
-target_method net.minecraft.client.gui.GuiGraphics vLine (IIII)V
-	param 3 argb_color
-
-target_method net.minecraft.client.gui.GuiGraphics fill (IIIII)V
-	param 4 argb_color
-
-target_method net.minecraft.client.gui.GuiGraphics fill (Lcom/mojang/blaze3d/pipeline/RenderPipeline;IIIII)V
-	param 5 argb_color
-
-target_method net.minecraft.client.gui.GuiGraphics fillGradient (IIIIII)V
-	param 4 argb_color
-	param 5 argb_color
-
-target_method net.minecraft.client.gui.GuiGraphics renderOutline (IIIII)V
-	param 4 argb_color
-
-target_method com.mojang.blaze3d.vertex.VertexConsumer setColor (I)Lcom/mojang/blaze3d/vertex/VertexConsumer;
-	param 0 argb_color
-
-target_method net.minecraft.network.chat.TextColor fromRgb (I)Lnet/minecraft/network/chat/TextColor;
-	param 0 argb_color
-
-target_method net.minecraft.network.chat.Style withColor (I)Lnet/minecraft/network/chat/Style;
-	param 0 argb_color
-
-target_method net.minecraft.network.chat.MutableComponent withColor (I)Lnet/minecraft/network/chat/MutableComponent;
-	param 0 argb_color
+target_annotation net.minecraft.server.level.TicketType$Flags chunk_ticket_type_flags
 
 group int cursor_shape
 	org.lwjgl.glfw.GLFW.GLFW_ARROW_CURSOR
@@ -277,7 +_,6 @@
 
 #: Flags used in {@link net.minecraft.entity.mob.VexEntity#VEX_FLAGS VEX_FLAGS}.
 group int vex_flags
-	@flags
 	net.minecraft.world.entity.monster.Vex.FLAG_IS_CHARGING
 
 target_method net.minecraft.world.entity.monster.Vex getVexFlag (I)Z
@@ -1569,6 +_,12 @@
 target_method net.minecraft.nbt.TagTypes getType (I)Lnet/minecraft/nbt/TagType;
 	param 0 nbt_type_id
 
+target_method net.minecraft.nbt.ListTag wrapIfNeeded (BLnet/minecraft/nbt/Tag;)Lnet/minecraft/nbt/Tag;
+	param 0 nbt_type_id
+
+target_method net.minecraft.nbt.ListTag identifyRawElementType ()B
+	return nbt_type_id
+
 group int max_string_length
 	net.minecraft.network.FriendlyByteBuf.MAX_STRING_LENGTH
 	net.minecraft.network.FriendlyByteBuf.MAX_COMPONENT_STRING_LENGTH
@@ -1585,6 +_,19 @@
 target_method net.minecraft.network.FriendlyByteBuf writeUtf (Ljava/lang/String;I)Lnet/minecraft/network/FriendlyByteBuf;
 	param 1 max_string_length
 
+group long max_nbt_size
+	net.minecraft.nbt.NbtAccounter.DEFAULT_NBT_QUOTA
+	net.minecraft.nbt.NbtAccounter.UNCOMPRESSED_NBT_QUOTA
+
+group int max_stack_depth
+	net.minecraft.nbt.NbtAccounter.MAX_STACK_DEPTH
+
+target_method net.minecraft.nbt.NbtAccounter <init> (JI)V
+	param 0 max_nbt_size
+	param 1 max_stack_depth
+target_method net.minecraft.nbt.NbtAccounter create (J)Lnet/minecraft/nbt/NbtAccounter;
+	param 0 max_nbt_size
+
 target_method net.minecraft.network.VarInt getByteSize (I)I
 	return max_var_int_length
 
@@ -1718,18 +_,9 @@
 
 target_annotation net.minecraft.world.level.block.Block$UpdateFlags set_block_state_flags
 
-target_method net.minecraft.world.level.Level sendBlockUpdated (Lnet/minecraft/core/BlockPos;Lnet/minecraft/world/level/block/state/BlockState;Lnet/minecraft/world/level/block/state/BlockState;I)V
+target_method net.minecraft.server.level.ServerLevel sendBlockUpdated (Lnet/minecraft/core/BlockPos;Lnet/minecraft/world/level/block/state/BlockState;Lnet/minecraft/world/level/block/state/BlockState;I)V
 	param 3 set_block_state_flags
 
-target_method net.minecraft.world.level.redstone.NeighborUpdater shapeUpdate (Lnet/minecraft/core/Direction;Lnet/minecraft/world/level/block/state/BlockState;Lnet/minecraft/core/BlockPos;Lnet/minecraft/core/BlockPos;II)V
-	param 4 set_block_state_flags
-
-target_method net.minecraft.server.level.WorldGenRegion setBlockState (Lnet/minecraft/core/BlockPos;Lnet/minecraft/world/level/block/state/BlockState;II)Z
-	param 2 set_block_state_flags
-
-target_method net.minecraft.world.level.chunk.ChunkAccess setBlockState (Lnet/minecraft/core/BlockPos;Lnet/minecraft/world/level/block/state/BlockState;I)Lnet/minecraft/world/level/block/state/BlockState;
-	param 2 set_block_state_flags
-
 group int trial_spawner_luminance
 	net.minecraft.world.level.block.entity.trialspawner.TrialSpawnerState$LightLevel.UNLIT
 	net.minecraft.world.level.block.entity.trialspawner.TrialSpawnerState$LightLevel.HALF_LIT
@@ -1828,3 +_,10 @@
 
 target_method net.minecraft.network.protocol.game.ClientboundLevelEventPacket getType ()I
 	return world_event
+
+target_method net.minecraft.world.item.AxeItem spawnSoundAndParticle (Lnet/minecraft/world/level/Level;Lnet/minecraft/core/BlockPos;Lnet/minecraft/world/entity/player/Player;Lnet/minecraft/world/level/block/state/BlockState;Lnet/minecraft/sounds/SoundEvent;I)V
+	param 5 world_event
+
+group int
+	@scope class net.minecraft.util.StringDecomposer
+	net.minecraft.util.StringDecomposer.REPLACEMENT_CHAR
